llvm/lib/Target/AMDGPU/SIPreEmitPeephole.cpp

```diff
@@ -615,11 +615,12 @@ void SIPreEmitPeephole::collectUnpackingCandidates(
   for (auto I = std::next(BeginMI.getIterator()); I != E; ++I) {
     MachineInstr &Instr = *I;
     uint16_t UnpackedOpCode = mapToUnpackedOpcode(Instr);
```

Comment:
I don't know why it's written this way, but you're making a set of redundant conditions worse. The mapToUnpackedOpcode call should be sunk and supersedes all of the other checks here (e.g., why is ib bothering checking isTerminator?)

Not sure what any of this has to do with adding the new case